docs/testplan/PMON-Services-Daemons-test-plan.md

```diff
@@ -209,25 +209,34 @@ Extend "check_pmon_daemon_status()" if there is any gap to follow the steps belo
    xcvrd                            RUNNING   pid 32, uptime 1:30:20
       ```
 
-3. Perform the following steps for the list of the expected daemon
-	a. Verify the daemon is running and check if all expected DB data is populated
-   b. Disable the critical process autorestart if it's enabled using [disable_and_enable_autorestart](https://github.com/Azure/sonic-mgmt/blob/73830827460750d88ba0a5de71bb5c79ad33c46e/tests/process_monitoring/test_critical_process_monitoring.py#L37)
-	c. Stop the daemon using `sudo docker exec pmon supervisorctl stop {daemon}`
-   d. Verify the daemon is not running using `sudo docker exec pmon supervisorctl status {daemon}` and check if all DB data is cleared 
-	e. Restart the daemon using `sudo docker exec pmon supervisorctl start {daemon}`
-   f. Verify the daemon is running using `sudo docker exec pmon supervisorctl status {daemon}` and check if all expected DB data is populated
-   h. Repeat from step c to step f for all exit signals which the daemon can exit with
-   i. Enable the critical process autorestart
+## Test case # 2 Verify each daemon status with DB data
+### Test objective
+Verify each daemon staus in PMON container using "check_pmon_daemon_status()" in "tests/common/platform/daemon_utils.py"
+Extend "check_pmon_daemon_status()" if there is any gap to follow the steps below.
+### Test steps
+Perform the following steps for the list of the expected daemon
+  - This flow will be able to be used for each daemon test (Test case #3 ~ #11) if it's applicable
+1. Verify the daemon is running and check if all expected DB data is populated
+2. Disable the critical process autorestart if it's enabled using [disable_and_enable_autorestart](https://github.com/Azure/sonic-mgmt/blob/73830827460750d88ba0a5de71bb5c79ad33c46e/tests/process_monitoring/test_critical_process_monitoring.py#L37)
+3. Stop the daemon using `sudo docker exec pmon supervisorctl stop {daemon}`
+4. Verify the daemon is not running using `sudo docker exec pmon supervisorctl status {daemon}` 
+5. Verify all DB data is cleared if any DB data has been populated while daemon was running 
+6. Restart the daemon using `sudo docker exec pmon supervisorctl start {daemon}`
+7. Verify the daemon is running using `sudo docker exec pmon supervisorctl status {daemon}`
+8. Verify all expected DB data is populated same as step 1.
+9. Repeat from step 3 to step 8 with sending the signal to kill the daemon for each exit signal being handled by the daemon
+10. Enable the critical process autorestart
 
-## Test case # 2 Data for chassisd daemon
+## Test case # 3 Data for chassisd daemon
```

Comment:
Rather than calling these "test cases", I think they should simply be called "Daemon-specific database data verification procedures". Then, step 1 above can simply read, `1. Verify the daemon is running and check if all expected DB data is populated (refer to daemon-specific database data verification procedure below)`.

Reply:
I removed the branch accidently so please review it from https://github.com/Azure/sonic-mgmt/pull/3213